core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -357,13 +359,22 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   def setLogLevel(logLevel: String) {
     val validLevels = Seq("ALL", "DEBUG", "ERROR", "FATAL", "INFO", "OFF", "TRACE", "WARN")
-    if (!validLevels.contains(logLevel)) {
+    // let's allow lowcase or mixed case too
+    if (!validLevels.contains(logLevel.toUpperCase)) {
```

Comment:
Can `validLevels` just become a `Set` in a `SparkContext` `object`? 
You can call `toUpperCase(Locale.ENGLISH)`, once, and save it in a val, and use the same value when checking the value and passing it to log4j, just for tidiness.

I don't think the `resetLogLevel` method is worth adding.